src/main/java/org/opensearch/security/tools/democonfig/SecuritySettingsConfigurer.java

```diff
@@ -151,15 +187,15 @@ static void writePasswordToInternalUsersFile(String adminPassword, String intern
     /**
      * Checks if security plugin is already configured. If so, the script execution will not continue.
      */
-    static void checkIfSecurityPluginIsAlreadyConfigured() {
+    void checkIfSecurityPluginIsAlreadyConfigured() {
         // Check if the configuration file contains the 'plugins.security' string
-        if (OPENSEARCH_CONF_FILE != null && new File(OPENSEARCH_CONF_FILE).exists()) {
-            try (BufferedReader br = new BufferedReader(new FileReader(OPENSEARCH_CONF_FILE, StandardCharsets.UTF_8))) {
+        if (this.installer.OPENSEARCH_CONF_FILE != null && new File(this.installer.OPENSEARCH_CONF_FILE).exists()) {
```

Comment:
nit: is `this` required everywhere here?

Would it make sense to move this method to the Installer or consider making getters for the values being accessed here?

Reply:
We do need to keep the method here since we split the installer class to be slightly modular. This class handles security related config settings. Given that we are using singleton instance, installer might not need this since all instances of SecuritySettingsConfigurer will still refer to same Installer instance. Now that I think out loud, I'll remove this.